### PR TITLE
o/snapstate: revert snaps with components

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This enables us to revert snaps with components, which results in just switching to the set of components that was installed when that revision of the snap was the current one.

### CODE REVIEW

File: built-snap/snapd_1337.2.65.2_amd64.snap.keep
Comment (pedronis) [inline built-snap/snapd_1337.2.65.2_amd64.snap.keep:1]: spurious addition?

File: overlord/snapstate/handlers_components.go [hunk 1]
```diff
@@ -424,6 +424,11 @@ func (m *SnapManager) doLinkComponent(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// during a revert, components should already be linked
+	if snapsup.Revert {
```
Comment (pedronis) [inline overlord/snapstate/handlers_components.go:428]: any reason not to skip adding the tasks instead in the first place?

**Review by pedronis — COMMENTED** [review]
thanks, question

File: overlord/snapstate/handlers_components.go (hunk 1 above)
Comment (andrewphelpsj) [reply overlord/snapstate/handlers_components.go]: Good question, I feel like I originally had a reason for taking this approach, but it doesn't seem relevant now. I've reworked this to not create the tasks.

**Review by pedronis — APPROVED** [review]
thank you

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, just a minor comment about the spread test

**Pull request merged**